src/Aspire.Hosting.Yarp/YarpNpmResourceExtensions.cs

```diff
@@ -0,0 +1,209 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Text;
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Yarp;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding YARP npm resources to the application model.
+/// </summary>
+[Experimental("ASPIREHOSTING001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+public static class YarpNpmResourceExtensions
+{
+    /// <summary>
+    /// Adds a YARP container configured to host static assets from a Node.js build.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <param name="workingDirectory">The working directory containing the Node.js project.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{YarpNpmResource}"/>.</returns>
+    /// <remarks>
+    /// This method creates a YARP container that serves static files built from a Node.js project.
+    /// The default configuration uses npm as the package manager, runs "npm install" and "npm run build",
+    /// and expects the output in a "dist" folder. Use the fluent configuration methods to customize these defaults.
+    /// </remarks>
+    [Experimental("ASPIREHOSTING001", UrlFormat = "https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+    public static IResourceBuilder<YarpNpmResource> AddYarpNpmApp(
+        this IDistributedApplicationBuilder builder,
+        [ResourceName] string name,
+        string workingDirectory)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+        ArgumentException.ThrowIfNullOrEmpty(workingDirectory);
+
+        var fullyQualifiedWorkingDirectory = Path.GetFullPath(workingDirectory, builder.AppHostDirectory)
+                                                 .TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+
+        var resource = new YarpNpmResource(name);
+        var options = new NodeStaticBuildOptions();
+        
+        // Set up initial container configuration (similar to AddYarp)
+        var resourceBuilder = builder.AddResource(resource)
+            .WithHttpEndpoint(name: "http", targetPort: 5000)
+            .WithImage("placeholder") // Temporary image, will be replaced by WithDockerfile
```

Comment:
This is supposed to be the Dockerfile that is generated based on the options correct?

Reply:
Yea, this should start with the yarp image though.

Reply:
Hmmmm...is this change functional without the correct base image? I'm trying to use it in an app and not sure if I need to call `WithDockerfile` in my app code to get it working or not.